variable/include/scipp/variable/variable.tcc

```diff
@@ -214,7 +214,9 @@ void VariableConceptT<T>::copy(const VariableConcept &other, const Dim dim,
     iterDims.resize(dim, delta);
 
   const auto &otherT = requireT<const VariableConceptT>(other);
-  auto otherView = otherT.valuesView(iterDims, dim, otherBegin);
+  // Pass invalid dimension to prevent slicing when non-range slice
+  auto otherView =
+      otherT.valuesView(iterDims, delta == 1 ? Dim::Invalid : dim, otherBegin);
```

Comment:
Now this will not work with variances again? Maybe better to do this in `concatenate` instead? Or does that not work?

Reply:
Doesn't easily work in `concatenate` because setting `Dim::Invalid` breaks many places in the copy where `Dimensions::contains` is called.